content/docs/use-cases/ci-and-ml.md

```diff
@@ -0,0 +1,172 @@
+the various stages of a data processing pipeline. Here is a non exhaustive list
+of a few other unique challenges:
+
+- Automatically re-train models upon changes
+- Systematically compare (and report) performance of different versions of a
+  model
+- Test data correctness when new datasets are introduced
+- Train a model which is computationally intensive or requires special
+  environment (e.g. GPU)
+
+[CI](https://en.wikipedia.org/wiki/Continuous_integration) is a core principal
+of nowadays software development in general and DevOps in particular. In a
+nutshell, it is aimed at frequent code, data, and model check-ins, automatic
+testing, and fast feedback. DVC provides a solid basis for automating model
+training and testing with constructs like pipelines and metrics. To streamline
+DVC into CI tools we've built a complementary tool, Continuous Machine Learning
+([CML](http://www.cml.dev)). Putting these tools together allows the bulding and
+**coding** of flows that can help tackling the aforementioned problems and many
+others. We will next dive a little deeper and outline how to harness these tools
+into CI flows that tackle the first two challenges.
+
+> The following code snippets are for demonstration purposes. Note that we use
+> GitHub Actions and
+> [workflow files](https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#creating-a-workflow-file)
+> in the following examples; the same ideas can be implemented also when using
+> other hosting solution that suports CI (e.g. GitLab).
+
+## Automatic model training
+
+Due to the complexity of the computations or special requirements of the
+training environment, for example, a training process can be very demanding.
+Nevertheless, it is very much desired to re-train whenever new data is available
+and/or changes were made to the model itself. Similarly, a test suite of a
+project is very involved and complex. CI is used to perform the testing; let it
+be on various platforms and environments, or simply move the testing load from a
+developer’s local machine to a dedicated testing platform. Just like CI helps
+when it comes to runing complex test suite, it can also help when the need for
+training arises.
+
+![](/img/ci_for_ml_retrain_model.jpg) _Model training flow_
+
+Assuming that in a project’s repository a DVC pipeline is defined for training a
+model. You can ask your CI tool to call `dvc pull` and, voilà, it now has access
+to the _right_ data needed for training. The reproducing pipeline can be
+arbitrary complicated and comprise of many steps; regardless of all, `dvc repro`
+will yield a newly trained model. This new model will be tracked and pushed to
+the remote cache (note `dvc push`) and will be "linked" to the PR.
+
+```yml
+# .github/workflows/train_model.yml
+
+name: Retrain model
+on:
+  pull_request:
+    types: synchronize
+
+jobs:
+  train:
+    runs-on: ubuntu-latest
```

Comment:
I presume that this is showcase for user, so that one can reproduce it by her/him self. In that case I would recommend sticking to particular version of system, so that we eliminate possibility of some dependencies/ environmental setup being different among images.

Reply:
@pared do you mean the version of Ubuntu? As the disclaimer states at the beginning, the snippets are for demonstration purposes - I expect readers to be inspired by this and work out the details as per their need. Therefore, although I totally understand that versions have to be included in the environment's definitions, I don't think it is important here. 

See my other comment. I'll mention you there :)

Reply:
Yeah, I guess I did look at it too much from our tests/benchmarks perspective - where we don't want any unnecessary changes. In this case that makes perfect sense to use the latest version.